collect/collect.go

```diff
@@ -571,6 +573,9 @@ func (i *InMemCollector) ProcessSpanImmediately(sp *types.Span, keep bool, sampl
 func (i *InMemCollector) dealWithSentTrace(keep bool, sampleRate uint, spanCount uint, sp *types.Span) {
 	if i.Config.GetAddRuleReasonToTrace() {
 		sp.Data["meta.refinery.reason"] = "late"
+
+	if i.Config.GetStressReliefConfig().Mode != "" {
```

Comment:
Can we use the same `if i.hostname !=nil` logic that is used in the `send` function?